devicetypes/smartthings/zwave-radiator-thermostat.src/zwave-radiator-thermostat.groovy

```diff
@@ -0,0 +1,262 @@
+								[value: 74, color: "#44b621"],
+								[value: 84, color: "#f1d801"],
+								[value: 95, color: "#d04e00"],
+								[value: 96, color: "#bc2323"]
+						]
+				)
+			}
+		}
+
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label: 'Battery:\n${currentValue}%', unit: "%"
+		}
+		standardTile("refresh", "command.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "refresh", label: 'refresh', action: "refresh.refresh", icon: "st.secondary.refresh-icon"
+		}
+		main "thermostat"
+		details(["thermostat", "battery", "refresh"])
+	}
+}
+
+def initialize() {
+	sendEvent(name: "checkInterval", value: 4 * 60 * 60 + 24 * 60 , displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	sendEvent(name: "supportedThermostatModes", value: thermostatSupportedModes, displayed: false)
+	response(refresh())
+}
+
+def installed() {
+	initialize()
+}
+
+def updated() {
+	initialize()
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	} else {
+		log.warn "${device.displayName} - no-parsed event: ${description}"
+	}
+	log.debug "Parse returned: ${result}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		log.debug "SecurityMessageEncapsulation into: ${encapsulatedCommand}"
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "unable to extract secure command from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def value = cmd.batteryLevel == 255 ? 1 : cmd.batteryLevel
+	def map = [name: "battery", value: value, unit: "%"]
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatmodev2.ThermostatModeReport cmd) {
+	def map = [name: "thermostatMode", data:[supportedThermostatModes: thermostatSupportedModes]]
+	switch (cmd.mode) {
+		case 1:
+			map.value = "heat"
+			break
+		case 11:
+			map.value = "emergency heat"
+			break
+		case 0:
+			map.value = "off"
+			break
+	}
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointReport cmd) {
+	createEvent(name: "heatingSetpoint", value: convertTemperatureIfNeeded(cmd.scaledValue, 'C', cmd.precision), unit: temperatureScale)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	def map = [name: "temperature", value: convertTemperatureIfNeeded(cmd.scaledSensorValue, 'C', cmd.precision), unit: temperatureScale]
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.warn "Unhandled command: ${cmd}"
+	[:]
+}
+
+def setThermostatMode(String mode) {
```

Comment:
Since turning off the thermostat when an unsupported mode is sent in to this command would be bad behavior, you should enforce a check on mode. Like here: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/f6f0b3ff7b31c83db0c78883f8383ca48ab04cb1/devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy#L366